sigstore/_verify.py

```diff
@@ -240,62 +236,45 @@ def verify(
 
         logger.debug("Successfully verified signature...")
 
-        # Get a base64 encoding of the signing key. We're going to use this in our Rekor query.
-        pub_b64 = base64.b64encode(
-            signing_key.public_bytes(
-                encoding=serialization.Encoding.PEM,
-                format=serialization.PublicFormat.SubjectPublicKeyInfo,
-            )
-        )
-
         # Retrieve the relevant Rekor entry to verify the inclusion proof and SET
-        uuids = self._rekor.index.retrieve.post(sha256_artifact_hash, pub_b64.decode())
+        entry = self._rekor.log.entries.retrieve.post(
+            signature.decode(),
+            sha256_artifact_hash,
+            base64.b64encode(certificate).decode(),
+        )
+        if entry is None:
+            return VerificationFailure(
+                reason="Rekor has no entry for these verification materials"
```

Comment:
While we're changing this, we probably could make this a bit more user friendly and explain what this means.